src/FieldWarning/Assets/Units/UnitDispatcher.cs

```diff
@@ -101,6 +104,11 @@ public class UnitDispatcher
                     this,
                     TargetTuple);
 
+            _armorComponent = new ArmorComponent(
+                _movementComponent,
+                platoon,
+                _movementComponent.Data.armorData);
```

Comment:
As mentioned above, if you want to instantiate a MonoBehaviour you can only do it with AddComponent<T>(). After you fix this (and in the future too) I recommend starting the game and getting some units to shoot each other to make sure your code works (in this case, that damage is applied correctly).